lib/src/localization.dart

```diff
@@ -26,18 +26,22 @@ class Localization {
     return translations == null ? false : true;
   }
 
-  String tr(String key, {List<String> args, String gender}) {
-    if (gender != null) return trGender(key, gender, args: args);
-    return _replaceArgs(_resolve(key), args);
+  String tr(String key, {List<String> args, Map<String, String> namedArgs, String gender}) {
```

Comment:
@aissat need to refactor in the future

Reply:
@Overman775  ??!!

Reply:
@aissat this code looking dirty, many params and `if (gender != null)` make code unreadable

Reply:
@3ace

Reply:
we could remove `trGender` method and replace `tr` method body with something like this
```
String res; 

if(gender != null) {
  res = _gender(key, gender: gender);
} else {
  res = _resolve(key);
}

res = _replaceNamedArgs(res, namedArgs);

return _replaceArgs(res, args);
```

Reply:
@3ace yep, look much better 👍